src/main/java/com/springweb/dv_spring_web_mongo/service/ProjectService.java

```diff
@@ -18,13 +20,33 @@
     @Autowired
     private ProjectRepository projectRepository;
 
-    public List<ProjectDTO> getAllProjects(String filterProjectName) {
-        List<Project> list;
+    public List<ProjectDTO> getAllProjects(String filterProjectName, int pageSize, int pageNumber) {
+        List<Project> list = null;
+        boolean paging = false;
+        boolean filteringByProjectName = true;
+
+        //condition switching
         if (filterProjectName == null) {
+            filteringByProjectName = false;
+        }
+        if (pageNumber > 0 && pageSize > 0) {
+            paging = true;
+        }
+
+        if (!filteringByProjectName && !paging) {
```

Comment:
Let's leave this solution as it is. 
But I believe You see that it is hard to support such approach.
For example what you will do if we will need to optionally sort the result (on db side)?
In such cases, a more preferable approach is to dynamically build a query to the database